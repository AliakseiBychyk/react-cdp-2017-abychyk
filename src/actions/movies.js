export const RECEIVE_MOVIES = 'RECEIVE_MOVIES'

export const receiveMovies = (movies) => ({
  type: RECEIVE_MOVIES,
  movies  
})

export const fetchMoviesJson = (criterion, query) => { 
  const url = `https://api.themoviedb.org/3/search/${criterion}?api_key=${api_key}&query=${query}&include_adult=true`   

  return fetch(url, {
    method: 'GET'
  }).then(resp => resp.json())
    .then(data => {
      return criterion === 'movie'
        ? data.results
        : data.results[0].known_for
    })
    .catch(console.error)
}