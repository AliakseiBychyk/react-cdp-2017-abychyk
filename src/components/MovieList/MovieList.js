import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import MoviePreview from '../MoviePreview/MoviePreview'
import styles from './MovieList.css'

const MovieList = ({movies, onMovieClick}) => (
  <div className={styles.movieList}>
    {movies.map((movie, index) =>
      <MoviePreview
        key={index}
        onClick={onMovieClick}
        {...movie}
      />
    )}
  </div>
)

const mapStateToProps = (state) => {
  return {
    movies: state.movies
  }
}

export default connect(mapStateToProps)(MovieList)
