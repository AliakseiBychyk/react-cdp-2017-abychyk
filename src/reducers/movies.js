import { RECEIVE_MOVIES } from '../actions/movies'

export default (state = [], action) => {
  switch (action.type) {
    
    case RECEIVE_MOVIES:
      return action.movies

    default:
      return state
  }
}