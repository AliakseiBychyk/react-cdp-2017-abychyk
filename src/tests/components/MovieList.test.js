import React from 'react'
import { shallow } from 'enzyme'
import MovieList from '../../components/MovieList/MovieList'
import movies from '../fixtures/testingMovieQuery.json'

test('should render MovieList with movies', () => {
  const wrapper = shallow(<MovieList movies={movies}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should render MovieList with empty movie array', () => {
  const wrapper = shallow(<MovieList movies={[]} />)
  expect(wrapper).toMatchSnapshot()
})