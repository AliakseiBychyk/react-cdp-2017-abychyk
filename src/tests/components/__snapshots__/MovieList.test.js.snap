// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieList with empty movie array 1`] = `
<div
  className="movieList"
/>
`;

exports[`should render MovieList with movies 1`] = `
<div
  className="movieList"
>
  <MoviePreview
    category="Dramas"
    director="James Cameron"
    key="0"
    mediatype={0}
    poster="http://netflixroulette.net/api/posters/1181461.jpg"
    rating="3.9"
    release_year="1997"
    runtime="194 min"
    show_cast="Leonardo DiCaprio, Kate Winslet, Billy Zane, Kathy Bates, Frances Fisher, Gloria Stuart, Bill Paxton, Bernard Hill, David Warner, Victor Garber"
    show_id={1181461}
    show_title="Titanic"
    summary="This cinematic spectacle -- winner of 11 Oscars -- follows society girl Rose and penniless artist Jack as they fall in love just as tragedy strikes."
    unit={352}
  />
  <MoviePreview
    category="Action & Adventure"
    director="James Cameron"
    key="1"
    mediatype={0}
    poster="http://netflixroulette.net/api/posters/1032625.jpg"
    rating="3.9"
    release_year="1984"
    runtime="107 min"
    show_cast="Arnold Schwarzenegger, Michael Biehn, Linda Hamilton, Paul Winfield, Lance Henriksen, Rick Rossovich, Bess Motta, Earl Boen, Dick Miller, Shawn Schepps"
    show_id={1032625}
    show_title="The Terminator"
    summary="In the post-apocalyptic future, tyrannical machines teleport a cyborg to 1984 to a kill a woman whose unborn son is destined to lead the human rebels."
    unit={380}
  />
  <MoviePreview
    category="Action & Adventure"
    director="James Cameron"
    key="2"
    mediatype={0}
    poster="http://netflixroulette.net/api/posters/60028202.jpg"
    rating="3.9"
    release_year="1991"
    runtime="137 min"
    show_cast="Arnold Schwarzenegger, Linda Hamilton, Robert Patrick, Edward Furlong, Earl Boen, Joe Morton, S. Epatha Merkerson, Castulo Guerra, Danny Cooksey, Jenette Goldstein"
    show_id={60028202}
    show_title="Terminator 2: Judgment Day"
    summary="A reprogrammed Terminator arrives from the future to protect young John Connor from a relentless shape-shifting cyborg in this sequel to Terminator. On the run, John and his guardian seek John's mother, Sarah, who is locked away in an insane asylum."
    unit={1001}
  />
</div>
`;
